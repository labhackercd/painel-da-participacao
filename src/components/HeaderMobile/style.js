/* eslint-disable import/prefer-default-export */
import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
  root: {
    alignItems: 'flex-end',
  },
  paper: {
    backgroundColor: '#363636',
    borderRadius: '15px 15px 0 0',
    color: 'white',
    margin: 0,
    maxWidth: '100%',
    width: '100%',
    '& .picker-container .picker-column .picker-item.picker-item-selected': {
      fontFamily: 'Open Sans',
      color: (props) => props.colors.borderColor,
      fontWeight: 700,
      fontSize: '14px',
    },
    '& .picker-container .picker-column .picker-item': {
      fontFamily: 'Open Sans',
      color: '#F0F0F0',
      fontSize: '14px',
    },
  },
  box: {
    background: (props) => props.colors.borderColor,
    borderRadius: '15px 15px 0 0',
    color: '#121212',
    height: '44px',
    display: 'flex',
    justifyContent: 'space-between',
  },
  typhography: {
    color: '#121212',
    fontWeight: 'bold',
    fontFamily: 'Open Sans',
    textTransform: 'capitalize',
    padding: '13px',
    fontSize: '11px',
  },
  selectMobile: {
    fontWeight: 700,
    padding: '10px',
    border: '2px solid',
    borderRadius: '5px',
    cursor: 'pointer',
  },
  selectMobileEnabled: {
    backgroundColor: 'transparent',
    color: '#FFFFFF',
    borderColor: (props) => props.colors.borderColor,
  },
  selectMobileDisabled: {
    backgroundColor: theme.palette.gray.tundora,
    color: 'gray',
    borderColor: 'gray',
  },
}));
