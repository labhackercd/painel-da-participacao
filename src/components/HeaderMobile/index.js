/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { useState } from 'react';
import dynamic from 'next/dynamic';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import Slide from '@material-ui/core/Slide';
import { Box, Typography } from '@material-ui/core';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import { useStyles } from './style';

import * as APPLICATION_CONSTANTS from '../../utils/constants/index';

const Picker = dynamic(() => import('react-mobile-picker'), {
  ssr: false,
});

const Transition = React.forwardRef(
  // eslint-disable-next-line prefer-arrow-callback
  function Transition(props, ref) {
    return <Slide data-testid="slide" direction="up" ref={ref} {...props} />;
  },
);

export default function HeaderMobile(props) {
  const colors = props.headerColors;

  const classes = useStyles({ colors });
  const [open, setOpen] = useState(false);
  const [selectedYear, setSelectedYear] = useState('Todo o período');
  const [month, setMonth] = useState('Todos os meses');
  const [selectMonthDisabled, setSelectMonthDisabled] = useState(true);
  const [buttonType, setButtonType] = useState('period');
  const [valueGroups, setValueGroups] = useState({});
  // eslint-disable-next-line max-len
  const rangeOfYears = (start, end) => (Array(end - start + 1).fill(start).map((year2, index) => year2 + index)).reverse();
  const yearsRange = rangeOfYears(props.initialYear, APPLICATION_CONSTANTS.CURRENT_YEAR);
  const [optionGroups, setOptionGroups] = useState({
    title: [],
  });
  const handleOpen = (type) => {
    setButtonType(type);
    if (type === 'period') {
      const array = ['Todo o período', ...yearsRange];
      setOptionGroups({
        title: array,
      });
    } else if (type === 'months') {
      const array = ['Todos os meses', ...APPLICATION_CONSTANTS.MONTHS_LIST.map((monthItem) => monthItem)];
      setOptionGroups({
        title: array,
      });
    }
    setOpen(true);
  };
  const handleClose = () => setOpen(false);

  const handleChangeYear = () => {
    setSelectedYear(valueGroups.title);
    if (valueGroups.title === 'Todo o período') {
      props.handlePeriodChange('0', '0');
      setSelectMonthDisabled(true);
    } else {
      setSelectMonthDisabled(false);
      props.handlePeriodChange('0', valueGroups.title);
    }
  };

  const handleChangeMonth = () => {
    setMonth(valueGroups.title);
    props.handlePeriodChange(optionGroups.title.indexOf(valueGroups.title, 0), selectedYear);
  };

  const selectFunction = () => {
    if (buttonType === 'period') {
      setMonth('Todos os meses');
      handleChangeYear();
    } else if (buttonType === 'months') {
      handleChangeMonth();
    }
    setOpen(false);
  };

  const handleChange = (name, value) => {
    setValueGroups(() => ({
      [name]: value,
    }));
  };

  return (
    <div>
      <Box display="flex">
        <button
          onClick={() => handleOpen('period')}
          type="button"
          variant="contained"
          style={{ marginRight: '19px', display: 'flex', alignContent: 'center' }}
          className={`${classes.selectMobile} ${classes.selectMobileEnabled}`}
        >
          <Typography alignSelf="center" alig>
            {selectedYear}
          </Typography>
          <ArrowDropDownIcon
            style={{
              color: '#FFFFFF',
              width: '24px',
              height: '24px',
              padding: 0,
              right: 0,
            }}
          />
        </button>
        <button
          onClick={() => selectedYear !== 'Todo o período' && handleOpen('months')}
          type="button"
          style={{ display: 'flex', alignContent: 'center' }}
          variant="contained"
          className={`${classes.selectMobile} ${selectMonthDisabled ? classes.selectMobileDisabled : classes.selectMobileEnabled}`}
        >
          <Typography alignSelf="center" alig>
            {month}
          </Typography>
          <ArrowDropDownIcon
            style={{
              color: '#FFFFFF',
              width: '24px',
              height: '24px',
              padding: 0,
              right: 0,
            }}
          />
        </button>
      </Box>
      <Dialog
        TransitionComponent={Transition}
        onClose={handleClose}
        open={open}
        classes={{ container: classes.root, paper: classes.paper }}
      >
        <Box className={classes.box}>
          <Button onClick={handleClose} className={classes.typhography}>Cancelar</Button>
          <Button onClick={() => selectFunction()} className={classes.typhography}>
            Confirmar
          </Button>
        </Box>
        <Picker
          optionGroups={optionGroups}
          valueGroups={valueGroups}
          onChange={handleChange}
        />
      </Dialog>
    </div>
  );
}
