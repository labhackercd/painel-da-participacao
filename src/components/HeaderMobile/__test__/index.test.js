/* eslint-disable no-unused-expressions */

import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent } from '@testing-library/dom';
import { screen, render } from '@testing-library/react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import HeaderMobile from '../index';
import MockTheme from '../../../mocks/theme/mockTheme';

const headerColors = {
  borderColor: '#DA7F0B',
  button: {
    main: '#DA7F0B',
    hover: '#C47209',
  },
  toolTipBackground: '14D768',
};

it('snapshot should not have changes', () => {
  const component = shallow(
    <MockTheme>
      <HeaderMobile
        initialYear="2021"
        headerColors={headerColors}
      />
    </MockTheme>,
  );
  expect(component.exists()).toEqual(true);
  expect(component).toMatchSnapshot();
});

test('Test if Header Mobile renders without crash', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MockTheme>
      <HeaderMobile
        initialYear="2021"
        headerColors={headerColors}
      />
    </MockTheme>, div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

test('test period button', () => {
  const text = 'Todo Per√≠odo';
  render(
    <MockTheme>
      <HeaderMobile
        initialYear="2021"
        headerColors={headerColors}
      />
    </MockTheme>,
  );

  const button = screen.getByText(text);

  fireEvent.click(button);

  expect(screen.getAllByText(text)).toBeVisible;
});

test('test months button', () => {
  const text = 'Todos os meses';
  render(
    <MockTheme>
      <HeaderMobile
        initialYear="2021"
        headerColors={headerColors}
      />
    </MockTheme>,
  );

  const button = screen.getByText(text);

  fireEvent.click(button);

  expect(screen.getAllByText(text)).toBeVisible;
});
