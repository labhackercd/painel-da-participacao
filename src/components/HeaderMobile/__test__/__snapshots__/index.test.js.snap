// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should not have changes 1`] = `ShallowWrapper {}`;
